### PR TITLE
Add rollback after slem migration

### PR DESCRIPTION
**Pull request by sofiasyria**
- Related ticket: https://progress.opensuse.org/issues/157096
- Verification runs: https://openqa.suse.de/tests/overview?build=sofiasyria%2Fos-autoinst-distri-opensuse%23slem_rollback&version=6.0&distri=sle-micro
- https://openqa.suse.de/tests/13846111

### CODE REVIEW

**Comment by sofiasyria** [comment]
WIP label is used because of temporary commending out some schedule modules so that VRs can pass

**Review by rakoenig — APPROVED** [review]
LGTM

File: lib/version_utils.pm [hunk 1]
```diff
@@ -744,6 +745,21 @@ sub check_os_release {
...
+Returns 1 (true) if os release version matches the one passed as arguement.
+If no arguements are given, the function will compare the os release version
+in /etc/os-release file with "VERSION" var.
+
+=cut
+
+sub verify_os_version {
+    my ($version, $os_release_file) = @_;
+    $version //= get_var("VERSION");
+    $os_release_file //= '/etc/os-release';
```
Comment (ricardobranco777) [inline lib/version_utils.pm:759]: You're not using this variable.

Comment (sofiasyria) [reply lib/version_utils.pm:759]: Right, too sloppy copying pasting. Thanks

**Review by ricardobranco777 — APPROVED** [review]

File: tests/microos/rollback.pm [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright 2016-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test downgrading to previous system version after
```
Comment (lemon-suse) [inline tests/microos/rollback.pm:6]: I think this is not downgrade, it is better to change it to 'rollback'. How about 'Test rollback to original system ...'

Comment (jknphy) [reply tests/microos/rollback.pm:6]: yes, rollback is to return to previous state, it is more clear, upgrade and downgrade are ambiguous terms.

Comment (sofiasyria) [reply tests/microos/rollback.pm]: Done

**Review by lemon-suse — APPROVED** [review]

**Comment by alvarocarvajald** [comment]
This broke our tests in QAM: https://openqa.suse.de/tests/13949214#step/zypper_migration/4

File: tests/migration/online_migration/zypper_migration.pm [hunk 3]
```diff
@@ -14,21 +14,17 @@ use testapi;
...
-    my $target_version = get_var("TARGET_VERSION", get_required_var("VERSION"));
-    assert_script_run("grep VERSION= /etc/os-release | grep $target_version");
-}
-
 sub run {
     my $self = shift;
     select_console 'root-console';
-
+    # This multi-naming of the same var should be addressed in poo#157873
+    my $version_to_verify = is_sle_micro ? get_var("FROM_VERSION") : get_var(("HDDVERSION"), get_required_var("ORIGIN_SYSTEM_VERSION"));
```
Comment (alvarocarvajald) [inline tests/migration/online_migration/zypper_migration.pm:26]: Breakage is introduced in this line.

The test is doing an rolling migration of a cluster node. Systems start on 15-SP{n} and get migrated by `zypper_migration` to 15-SP{n+1}. As such, when `verify_os_version` is called, the actual OS version is `ORIGIN_SYSTEM_VERSION` not `HDDVERSION` but both are set.

Comment (sofiasyria) [reply tests/migration/online_migration/zypper_migration.pm:26]: I am removing the verify_os_version as there are many tests that use different var naming for the same thing

**Pull request merged**